addons/mail/populate/mail_channel.py

```diff
@@ -0,0 +1,60 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models
+from odoo.tools import populate
+
+
+class Channel(models.Model):
+    _inherit = "mail.channel"
+    _populate_dependencies = ["mail.message", "res.partner"]
+
+    _populate_sizes = {
+        'small': 100,
+        'medium': 1.105e5,
+        'large': 5.1e5,
+    }
+
+    def get_channel_probability(self):
+        return [0.1, 0.9]
```

Comment:
It would be more clear if this was a dict with keys for each type instead of an array.
Or just `get_channel_type` could return both the channel and its probability?